pkg/distributors/contracts/MultiDistributor.sol

```diff
@@ -290,11 +291,19 @@ contract MultiDistributor is IMultiDistributor, ReentrancyGuard, MultiDistributo
             // Fixed point (FP) multiplication between a non-FP (time) and FP (rate) returns a non-FP result.
             uint256 leftoverTokens = FixedPoint.mulDown(remainingTime, distribution.paymentRate);
             // Fixed point (FP) division of two non-FP values we get a FP result.
-            distribution.paymentRate = FixedPoint.divDown(amount.add(leftoverTokens), duration);
+            paymentRate = uint192(FixedPoint.divDown(amount.add(leftoverTokens), duration));
         }
 
-        distribution.lastUpdateTime = block.timestamp;
-        distribution.periodFinish = block.timestamp.add(duration);
+        require(paymentRate < type(uint192).max, "PAYMENT_RATE_OVERFLOW");
```

Comment:
I think for this require to do anything we need to not cast the result of `divDown` to 192, as the casting will force the value to fit in the type: see [here](https://docs.soliditylang.org/en/v0.8.10/types.html?highlight=higher-order%20bits%20are%20cut%20off#explicit-conversions).

Reply:
Thanks, I missed removing these casts when adding the overflow check here.